api/stats.js

```diff
@@ -1,31 +1,73 @@
 import { config } from './../config.js';
 import { SnowflakeUtil, MessageEmbed } from "discord.js";
 
+const removeParentheses = word => word.replace(/\(|\)/g, '');
 export class Stats {
     constructor(discordClient) {
         this.client = discordClient;
         this.stats = {};
     }
 
-    initializeStats(prev_message) {
-        /*
-        This is a placeholder.
-        We have some limitations on how many messages we can load per request
-        we can have daily pulls, use the latest message as the last source of truth
-        and update with new stats
-        let stats = {
-            username1: {
-                messages: 5, 
-                words: 20, 
-                letters: 150
+    /*
+        Returns a snowflake which represents the time x days in the past
+        https://discord.com/developers/docs/reference#snowflakes
+    */
+    _generatDaySnowflake(days) {
+        let d = new Date();
+        d.setHours(0, 0, 0, 0);
+        d.setDate(d.getDate() + days);
+        return SnowflakeUtil.generate(d);
+    }
+
+    async _getStatsChannel() {
+        let dominationGuild = this.client.guilds.resolve(config.GUILD_ID);
+        return dominationGuild.channels.cache.find(
+            (channel) => channel.name === config.STATS_CHANNEL
+        );
+    }
+
+    async _computeWeeklyStats() {
+        let sevenDaysAgo = this._generatDaySnowflake(-7);
+        let statsChannel = await this._getStatsChannel();
+
+        let messages = await statsChannel.messages.fetch({ limit: 100, after: sevenDaysAgo });
+        let collectedDays = new Set();
+        let stats = {};
+
+        for (let message of messages.values()) {
+            let isBot = message.author.username === 'elprimobot';
+            let isDaily = message.content.indexOf('Daily') > -1;
+            let day = new Date(message.createdTimestamp);
+            let key = `${day.getDay()}/${day.getMonth()}/${day.getFullYear()}`;
+            // we only take the latest stat posted for the bot that day
+            if (!isBot || !isDaily || collectedDays.has(key)) {
+                continue;
+            }
+            collectedDays.add(key);
+
+            let fields = message.embeds[0].fields;
+            let firstColumn = fields[0].value.split('\n');
+            let secondColumn = fields[2].value.split('\n');
+            console.log(firstColumn, secondColumn);
```

Comment:
Remove console log, let's keep it clean.

Reply:
i removed it, however, i think it's helpful sometimes to have these logs when you're debugging a problem in the server.

Reply:
There's no effect on this. Neither a point on why removing it makes it clean.